client/src/main/java/io/pravega/client/segment/impl/EventSegmentInputStream.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.segment.impl;
+
+import io.pravega.client.stream.EventStreamWriter;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines a InputStream for a single segment.
+ * Once created the offset must be provided by calling setOffset.
+ * The next read will proceed from this offset. Subsequent reads will read from where the previous
+ * one left off. (Parallel calls to read data will be serialized)
+ * Get offset can be used to store a location to revert back to that position in the future.
+ */
+public interface EventSegmentInputStream extends AutoCloseable {
```

Comment:
Can we simplify this name to `EventInputStream`?  Ultimately, the unit of input is event. Also, `EventSegment` sounds odd.

Reply:
Sounds good to me. I missed that this doesn't extend from an `InputStream`. We should call this `*Reader`